### ISSUE TITLE
Zone transfer error: REFUSED / NOTAUTH

### ISSUE THREAD

**Issue by blshkv** [mschwager/fierce#26]
I'm using the latest git and facing the following issue:
```
fierce --domain itdefence.asia
NS: ns-77-a.gandi.net. ns-187-b.gandi.net. ns-216-c.gandi.net.
SOA: ns1.gandi.net. (173.246.98.2)
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.5/fierce", line 11, in <module>
    load_entry_point('fierce==1.2.0', 'console_scripts', 'fierce')()
  File "/usr/lib64/python3.5/site-packages/fierce/fierce.py", line 389, in main
    fierce(**vars(args))
  File "/usr/lib64/python3.5/site-packages/fierce/fierce.py", line 248, in fierce
    zone = zone_transfer(master_address, domain)
  File "/usr/lib64/python3.5/site-packages/fierce/fierce.py", line 120, in zone_transfer
    return dns.zone.from_xfr(dns.query.xfr(address, domain))
  File "/usr/lib64/python3.5/site-packages/dns/zone.py", line 1063, in from_xfr
    for r in xfr:
  File "/usr/lib64/python3.5/site-packages/dns/query.py", line 611, in xfr
    raise TransferError(rcode)
dns.query.TransferError: Zone transfer error: NOTAUTH
```

I have also seen the "REFUSED" error message with an another domain.

Please also note that I'm using the latest git of dnspython:
```dev-python/dnspython-1.16.0_pre20170831```

**Comment by mschwager** [comment]
https://github.com/mschwager/fierce/commit/611f833603852af2ab57648712da0cb5a15b6a8c should fix your issue. However, `dnspython==1.16.0` is not officially supported, it's not even in PyPI yet. `dns.query.TransferError` doesn't exist in the latest release (`1.15.0`), so there's no way to test that that commit fixes your issue, although I suspect it does. There could be other underlying issues, though, since `fierce` doesn't expect that version of `dnspython`. I'd recommend using the latest officially released code.

I've also tagged https://github.com/mschwager/fierce/releases/tag/1.2.1-rc2.

**Comment by blshkv** [comment]
The problem with dnspython-1.15.0 is that it requires unsupported and vulnerable pycrypto. 
There is a pull request https://github.com/rthalley/dnspython/pull/290 with a patch however the patch can be applied against the latest git. I'll see if I can backport the patch.

Thank you for the fix!

**Comment by mschwager** [comment]
I see, thanks for the context! I will upgrade `fierce` to `dnspython==1.16.0` as soon as it's released!

**Comment by blshkv** [comment]
well, it was released last year. You have closed this bug and forgot about it probably?

https://github.com/rthalley/dnspython/releases/tag/v1.16.0

**Comment by mschwager** [comment]
Heads up @blshkv, I just released `1.3.0` (https://pypi.org/project/fierce/1.3.0/), which includes the bump to `dnspython` `1.16.0` :+1:
